### PR TITLE
fix: run urlify on frontend to render links

### PR DESCRIPTION
**Pull request by AnupNair08**
Markup is not stored on the database anymore. Urlify runs on the frontend while rendering messages. 
_**Note that this may cause all the old messages to appear in a markup format**_

### CODE REVIEW

File: src/views/Message/index.js [hunk 1]
```diff
@@ -24,7 +36,7 @@ export default function Message(props) {
             {!isMine && data.sender}
           </b>
           {/* {data.msg} */}
-          <div dangerouslySetInnerHTML={{ __html: data.msg }} />
+          <div dangerouslySetInnerHTML={{ __html: urlify(data.msg) }} />
```
[the remark this answers is not in the document]
Comment (yashshah1) [reply src/views/Message/index.js:39]: `(https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|www\.[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9]+\.[^\s]{2,}|www\.[a-zA-Z0-9]+\.[^\s]{2,})`

This works too.
[Source](https://stackoverflow.com/a/17773849/11362183)

**Comment by AnupNair08** [comment]
@yashshah1 
Worked on the following changes:
- Used a more better regular expression for URL matching. (The one suggested  in the earlier comment)
-  dangerouslySetInnerHTML is replaced with ```html-react-parse``` library to parse the message strings.

>

**Comment by yashshah1** [comment]
@AnupNair08 Why add an extra dependency for this use-case? Returning JSX from urlify if it is a URL, or text as is seems like a lighter approach.

In the case of a mixed string, where plaintext and URLs are present, what an approach could be is:
- let urlify return the _entire message div_

So in the case the message is something like: 
` Hey checkout www.google.com and www.bing.com`
urlify could return
```js
<div>
Hey checkout <a href="www.google.com" {...props}>www.google.com</a> and <a href="www.bing.com" {...props}>www.bing.com</a>
</div>
```
where the example above is JSX
@veddandekar @Pranav2612000 thoughts?

**Comment by AnupNair08** [comment]
@yashshah1 
```text.match()``` expects a callback function, the return value of which is directly replaced into our message string. If we were to return JSX then it tries to replace string with JSX. If we were to return a string from the callback, the entire message then is a unified string with markup which was parsed using the dependency.
This was the reason for going with the dependency.
Any possible workarounds?

**Comment by yashshah1** [comment]
Hi, There is just a little string manipulation required:
- Tokenise the string to be either a URL or plaintext.
- For the URL bit, convert to JSX
- For the text bit, let it be
- Render this mixed list.


See [this quick mockup](https://codepen.io/yashs/pen/gOPQNBW), covers quite a few corner cases, but I haven't been able to test it out extensively. But the example cited above works.

**Comment by AnupNair08** [comment]
@yashshah1 
Thank you for the mockup. Working on the suggested changes.

**Comment by yashshah1** [comment]
LGTM @AnupNair08 :)
